src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java

```diff
@@ -22,547 +22,540 @@
-	}
-
-	/**
-	 * This function tries to archive a task folder in Swift.
-	 * 
-	 * @param task     task in archive process
-	 * @param localDir task folder to be archived
-	 * @param swiftDir directory swift to archive new data
-	 * @return boolean representation, success (true) or failure (false) to archive
-	 */
-	private boolean archive(SapsImage task, String localDir, String swiftDir) {
-		LOGGER.info("Trying to archive task [" + task.getTaskId() + "] " + localDir + " folder with a maximum of "
-				+ MAX_ARCHIVE_TRIES + " archiving attempts");
-
-		File localFileDir = new File(localDir);
-
-		if (!localFileDir.exists() || !localFileDir.isDirectory()) {
-			LOGGER.error("Failed to archive task [" + task.getTaskId() + "]. " + localDir
-					+ " folder isn't directory or not exists");
-			return false;
-		}
-
-		for (int itry = 0; itry < MAX_ARCHIVE_TRIES; itry++) {
-			if (uploadFilesToSwift(task, localFileDir, swiftDir))
-				return true;
-		}
-
-		return false;
-	}
-
-	/**
-	 * This function tries upload task folder files to Swift.
-	 * 
-	 * @param task     task in archive process
-	 * @param localDir task folder to be archived
-	 * @param swiftDir directory swift to archive new data
-	 * @return boolean representation, success (true) or failure (false) to archive
-	 */
-	private boolean uploadFilesToSwift(SapsImage task, File localDir, String swiftDir) {
-		LOGGER.info("Trying to archive task [" + task.getTaskId() + "] " + localDir + " folder for swift");
-		for (File actualFile : localDir.listFiles()) {
-			if (!uploadFileToSwift(actualFile, swiftDir)) {
-				LOGGER.info("Failure in archiving file [" + actualFile.getAbsolutePath() + "]");
-				return false;
-			}
-		}
-
-		LOGGER.info("Upload to swift succsessfully done");
-		return true;
-	}
-
-	/**
-	 * This function tries upload a task folder file to Swift.
-	 * 
-	 * @param actualFile file to be uploaded
-	 * @param swiftDir   directory swift to archive
-	 * @return boolean representation, success (true) or failure (false) to archive
-	 */
-	private boolean uploadFileToSwift(File actualFile, String swiftDir) {
-		String containerName = getContainerName();
-		LOGGER.info("Trying to archive file [" + actualFile.getAbsolutePath() + "] for swift container ["
-				+ containerName + "] with a maximum of " + MAX_SWIFT_UPLOAD_TRIES + " uploading attempts");
-
-		for (int itry = 0; itry < MAX_SWIFT_UPLOAD_TRIES; itry++) {
-			try {
-				swiftAPIClient.uploadFile(containerName, actualFile, swiftDir);
-				return true;
-			} catch (Exception e) {
-				LOGGER.error("Error while uploading file " + actualFile.getAbsolutePath() + " to swift", e);
-			}
-		}
-
-		return false;
-	}
-
-	private String getContainerName() {
-		return properties.getProperty(SapsPropertiesConstants.SWIFT_CONTAINER_NAME);
-	}
+    private final Properties properties;
+    private final Catalog catalog;
+    private final SwiftAPIClient swiftAPIClient;
+    private ScheduledExecutorService sapsExecutor;
```

Comment:
shouldn't this `ScheduledExecutorService` var be declared final?

Reply:
Done